File: Source/OldCode/Archive/2007 Phasor Protocols/PhasorProtocols (VB)/ICommandCell.vb
'*******************************************************************************************************
'  ICommandCell.vb - Command cell interface
'  Copyright © 2008 - TVA, all rights reserved - Gbtc
'
'  Build Environment: VB.NET, Visual Studio 2008
'  Primary Developer: J. Ritchie Carroll, Operations Data Architecture [TVA]
'      Office: COO - TRNS/PWR ELEC SYS O, CHATTANOOGA, TN - MR 2W-C
'       Phone: 423/751-2827
'       Email: jrcarrol@tva.gov
'
'  Code Modification History:
'  -----------------------------------------------------------------------------------------------------
'  04/16/2005 - J. Ritchie Carroll
'       Initial version of source generated
'
'*******************************************************************************************************

''' <summary>This interface represents the protocol independent representation of an element of extended command frame data.</summary>
<CLSCompliant(False)> _
Public Interface ICommandCell

    Inherits IChannelCell

    Shadows ReadOnly Property Parent() As ICommandFrame

    Property ExtendedDataByte() As Byte

End Interface

File: Source/OldCode/Archive/2007 Phasor Protocols/PhasorProtocols/ICommandCell.cs
//*******************************************************************************************************
//  ICommandCell.cs
//  Copyright © 2009 - TVA, all rights reserved - Gbtc
//
//  Build Environment: C#, Visual Studio 2008
//  Primary Developer: James R Carroll
//      Office: PSO TRAN & REL, CHATTANOOGA - MR BK-C
//       Phone: 423/751-4165
//       Email: jrcarrol@tva.gov
//
//  Code Modification History:
//  -----------------------------------------------------------------------------------------------------
//  04/16/2005 - James R Carroll
//       Generated original version of source code.
//
//*******************************************************************************************************

using System;

namespace TVA.PhasorProtocols
{
    /// <summary>
    /// Represents a protocol independent interface representation of any kind of <see cref="ICommandFrame"/> cell.
    /// </summary>
    public interface ICommandCell : IChannelCell
    {
        /// <summary>
        /// Gets a reference to the parent <see cref="ICommandFrame"/> for this <see cref="ICommandCell"/>.
        /// </summary>
        new ICommandFrame Parent { get; set; }

        /// <summary>
        /// Gets or sets extended data <see cref="Byte"/> that represents this <see cref="ICommandCell"/>.
        /// </summary>
        byte ExtendedDataByte { get; set; }
    }
}
